app/lib/text/generator.rb

```diff
@@ -91,7 +91,7 @@ def components
         @components = @report.active_text_components(@opts)
         @components = components.shuffle
         @nil_priorities, @components = components.partition {|c| c.priority.nil? }
-        @components = components.sort_by {|c| Trigger.priorities[c.priority] }
+        @components = components.sort_by(&:priority_index)
```

Comment:
After looking at your second PR, I can see another shared responsibility: 3. Substitution of markup. If you decompose and reuse that part, you can be quite sure that the chatfuel controller can also deliver messages like: `It is 32°C right now.` There are tests for this functionality already, so you wouldn't need to test that again.